app/lib/features/bluetooth/logic/device_scan_cubit.dart

```diff
@@ -91,22 +89,23 @@ class DeviceScanCubit extends Cubit<DeviceScanState> {
   }
 
   void _onScanError(Object error) {
-    Log.err('Starting device scan failed', [ error ]);
+    logger.severe('Error during device discovery', error);
   }
 
   /// Mark a new device as known and switch to selected device state asap.
   Future<void> acceptDevice(BluetoothDevice device) async {
     assert(state is! DeviceSelected);
     try {
-      await _flutterBluePlus.stopScan();
+      await _stopScanning();
     } catch (e) {
       _onScanError(e);
       return;
     }
-    assert(!_flutterBluePlus.isScanningNow);
+
+    assert(!_manager.discovery.isDiscovering);
     emit(DeviceSelected(device));
     final List<String> list = settings.knownBleDev.toList();
-    list.add(device.platformName);
+    list.add(device.name);
```

Comment:
Missed that one ;). I can't think of a realistic use case where you have to differentiate between two of the same device. Using an device id here wouldn't help with that since all information the user has is the name. Additionally changing this behavior would either require users to re-add their devices after an update or add complexity in the form of duplicate code to maintain.

Reply:
The user can't add arbitrary device names, but they are only shown (and most likely only know) the name. Users _could_ ensure only their device is enabled when initially adding it, but this isn't really an intuitive UX I would want to endorse.

I wouldn't want to spend too much thought on it. This edge case would only occur when multiple devices of the exact same type take measurements at the same time in close proximity. I can't think of a place where this would be the case.

If you wanted to be extra sure you could auto-select the device with the strongest signal (e.g. closest to the phone), but I don't think that is really neccessary